### PR TITLE
Script for when archway rebuild is done

### LINKED ISSUE #731: BUG: Gnarled Forest pirate arch does not give shield
### Make sure you've done the following:

- [X] I have checked that this issue has not already been reported.

- [X] I have validated that this issue is not a syntax error of either MySQL or SQLite.

- [X] I have pulled the latest version of the main branch of DarkflameServer and have confirmed that the issue exists there.

### DarkflameServer Version

yes

### Issue Description

the rebuildable pirate arch at the ravine, LOT 6917, is supposed to give a temporary shield to the player upon the rebuild completing, see:

scripts\ai\GF\L_GF_ARCHWAY.lua

simply casts skill 863 on whoever rebuilt the arch upon completing it

https://youtu.be/5OrRMmr4aUU?list=PL9EFB8500E625ADC7&t=478

rather useless feature from a gameplay standpoint tbh

### Reproduction Steps

build arch

### Expected Behavior

get shielded

### Environment

gnarled forest

### PR DESCRIPTION
**Pull request by aronwk-aaron**
Resolves #731

### CODE REVIEW

File: dScripts/CppScripts.cpp [hunk 1]
```diff
@@ -493,6 +494,9 @@ CppScripts::Script* CppScripts::GetScript(Entity* parent, const std::string& scr
 		script = new GfApeSmashingQB();
 	else if (scriptName == "scripts\\zone\\PROPERTY\\GF\\L_ZONE_GF_PROPERTY.lua")
 		script = new ZoneGfProperty();
+	else if (scriptName == "") {
```
Comment (EmosewaMC) [inline dScripts/CppScripts.cpp:497]: What is the script name?

File: dScripts/GfArchway.cpp [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#include "GfArchway.h"
+#include "GameMessages.h"
```
Comment (EmosewaMC) [inline dScripts/GfArchway.cpp:2]: No use in game messages being included here

**Review by EmosewaMC — CHANGES_REQUESTED** [review]

File: dScripts/GfArchway.h [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include "CppScripts.h"
+
+class GfArchway : public CppScripts::Script {
+public:
+	void OnRebuildComplete(Entity* self, Entity* target) override;
+private:
+	uint32_t SHIELDING_SKILL = 863;
```
Comment (EmosewaMC) [inline dScripts/GfArchway.h:8]: We can also make these const

**Comment by EmosewaMC** [comment]
Fixes #731

**Comment by EmosewaMC** [comment]
I presume you've tested that the arch does give the skill, correct?

**Comment by aronwk-aaron** [comment]
> I presume you've tested that the arch does give the skill, correct?

yes, it has been tested, forgot to mention that

**Review by EmosewaMC — APPROVED** [review]

**Pull request merged**